auth/passwords.cc

```diff
@@ -46,7 +46,7 @@ sstring hash_with_salt(const sstring& pass, const sstring& salt) {
     return res;
 }
 
-const char* prefix_for_scheme(scheme c) noexcept {
+std::string_view prefix_for_scheme(scheme c) noexcept {
```

Comment:
the title of this commit reads

> change return type of prefix_for_scheme to std::string_view

but it does not explain why we want to make this change. i checked all callers of `prefix_for_scheme()`. both of them convert the return value of `prefix_for_scheme()` to `sstring` . and the constructor of `sstring` uses `basic_sstring(s.data(), s.size())` to construct an instance of `sstring` from `std::string_view`.  in `std::string_view` 's constructor of `string_view(const char*)`, it is able to use a `constexpr` variant of `strlen()`. so in addition to better semantics of "returning a string", i guess this change can also potentially improve the performance.

anyway, i'd prefer having a short explanation on "why". even it just claims that `std::string_view` is a better interface than `const char*`, because the latter is a pointer, and the former sends a message to human readers that we are returning a string from this function.